docs/r2dbc.md

```diff
@@ -0,0 +1,286 @@
+be ignored when connecting with the Cloud SQL Connector using IAM auth, leaving it empty will cause
+driver-level validations to fail.
+
+#### Mysql
+
+```java
+    // Set up ConnectionFactoryOptions
+ConnectionFactoryOptions options = ConnectionFactoryOptions.builder()
+    .option(DRIVER, "gcp")
+    .option(PROTOCOL, "mysql")
+    .option(PASSWORD, "password")
+    .option(USER, "mysql-iam-user@gmail.com")
+    .option(DATABASE, "my_db")
+    .option(HOST, "project:region:instance")
+    .option(ENABLE_IAM_AUTH, true)
+    .build();
+
+// Initialize connection pool
+ConnectionFactory connectionFactory = ConnectionFactories.get(options);
+ConnectionPoolConfiguration configuration = ConnectionPoolConfiguration
+    .builder(connectionFactory)
+    .build();
+
+this.connectionPool = new ConnectionPool(configuration);
+```
+
+#### Postgres
+
+```java
+// Set up ConnectionFactoryOptions
+ConnectionFactoryOptions options = ConnectionFactoryOptions.builder()
+    .option(DRIVER, "gcp")
+    .option(PROTOCOL, "postgresql")
+    .option(PASSWORD, "password")
+    .option(USER, "postgres-iam-user@gmail.com")
+    .option(DATABASE, "my_db")
+    .option(HOST, "project:region:instance")
+    .option(ENABLE_IAM_AUTH, true)
+    .build();
+
+// Initialize connection pool
+ConnectionFactory connectionFactory = ConnectionFactories.get(options);
+ConnectionPoolConfiguration configuration = ConnectionPoolConfiguration
+    .builder(connectionFactory)
+    .build();
+
+this.connectionPool = new ConnectionPool(configuration);
+```
+
+#### SQL Server
+
+Not Supported
+
+## Service Account Impersonation
+The Java Connector supports service account impersonation with the
+`TARGET_PRINCIPAL` option. When enabled, all API requests are made impersonating
+the supplied service account. The IAM principal must have the
+iam.serviceAccounts.getAccessToken permission or the role
+roles/iam.serviceAccounts.serviceAccountTokenCreator.
+
+#### Mysql
+
+```java
+// Set up ConnectionFactoryOptions
+ConnectionFactoryOptions options = ConnectionFactoryOptions.builder()
+    .option(DRIVER, "gcp")
+    .option(PROTOCOL, "mysql")
+    .option(PASSWORD, "password")
+    .option(USER, "mysql-iam-user@gmail.com")
+    .option(DATABASE, "my_db")
+    .option(HOST, "project:region:instance")
+    .option(ENABLE_IAM_AUTH, true)
+    .option(TARGET_PRINCIPAL, "mysql-iam-user@gmail.com")
+    .build();
+
+// Initialize connection pool
+ConnectionFactory connectionFactory = ConnectionFactories.get(options);
+ConnectionPoolConfiguration configuration = ConnectionPoolConfiguration
+    .builder(connectionFactory)
+    .build();
+
+this.connectionPool = new ConnectionPool(configuration);
+```
+
+#### Postgres
+
+```java
+// Set up ConnectionFactoryOptions
+ConnectionFactoryOptions options = ConnectionFactoryOptions.builder()
+    .option(DRIVER, "gcp")
+    .option(PROTOCOL, "postgresql")
+    .option(PASSWORD, "password")
+    .option(USER, "postgres-iam-user@gmail.com")
```

Comment:
Let's use a service account example instead of a human user.

Reply:
Fixed.